trino/client.py

```diff
@@ -524,8 +524,27 @@ def _map_to_python_type(cls, item: Tuple[Any, Dict]) -> Any:
                     "typeSignature": data_type["typeSignature"]["arguments"][0]["value"]
                 }
                 return [cls._map_to_python_type((array_item, raw_type)) for array_item in value]
+            if isinstance(value, dict):
+                raw_key_type = {
+                    "typeSignature": data_type["typeSignature"]["arguments"][0]["value"]
+                }
+                raw_value_type = {
+                    "typeSignature": data_type["typeSignature"]["arguments"][1]["value"]
+                }
+                return {
+                    cls._map_to_python_type((key, raw_key_type)): cls._map_to_python_type((value[key], raw_value_type))
+                    for key in value
+                }
             elif "decimal" in raw_type:
                 return Decimal(value)
+            elif "double" in raw_type:
+                if value == 'Infinity':
+                    return float("inf")
+                elif value == '-Infinity':
+                    return float("-inf")
+                elif value == 'NaN':
+                    return float("nan")
```

Comment:
We could use `math.inf` & `math.nan` since the minimum supported version is 3.7 now. Please define constants if you use `float(x)` style.